### PR TITLE
3.5.6 - 14386 - Don't show bogus 'module saved' confirmations if save was cancelled by user at a dialog.

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

**Comment by uckelman** [comment]
For thoroughness, I checked VASL to see if it uses ArchiveWriter directly. It doesn't, so it's very unlikely that anything else does.

**Review by uckelman — APPROVED** [review]

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java [hunk 1]
```diff
@@ -267,26 +267,31 @@ public void addFile(String fileName, byte[] content) {
...
-    if (isTempArchive) saveAs(notifyModuleManager);
-    else write(archive, notifyModuleManager);
+  public boolean save(boolean notifyModuleManager) throws IOException {
+    if (isTempArchive) {
+      return saveAs(notifyModuleManager);
+    }
+    else {
+      write(archive, notifyModuleManager);
+    }
+    return false;
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:286]: Shouldn't we return `true` after `write()`?

Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:286]: Is this the cause of the problem noted here? http://www.vassalengine.org/forum/viewtopic.php?p=67818#p67818

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:286]: Yep, had to make a new PR for it, so see #891

**Pull request merged**